black-cape/cast-iron-worker
# Add Postgres for Workflow tracking

This PR is to add Postgres to Cast Iron to allow for more robust workflow tracking.  Essentially, the database is used to track the file and metadata generated from an event and allows for that information to persist past the point of processing via Kafka.  Additional PR will follow for script to replay from DB to Kafka and for recovery options for pipeline interruptions.  With the introduction of the DB, filenames are now prepended with a uuid to prevent filename collision from multiple Cast Iron users.

Testing:
- Start Cast Iron with supporting Postgres DB (see corresponding branch in the cast-iron project for an example)
- Load a processing TOML
- Place a file to be processed in the inbox and verify that the file is processed properly
- Connect to the Postgres container with your tool of choice and verify a new row exists in the database (default is castiron with table name of files)
